### DISCUSSION TITLE
How do I make convolutional neural networks (CNN) in Mojo 🔥

### DISCUSSION THREAD

**Discussion by Eprahim-taha** [modular/modular#1511] [General]
I am a university student and I am doing research on convolutional neural networks (CNN) in Mojo 🔥, Actually I want to use (CNN) in robotics arm

**Comment by talhatahir-10xe** [comment]
See this it might be helpful
https://github.com/jiel/laplacian_filters_benchmark/tree/main

**Comment by hogepodge** [comment]
Hi @Eprahim-taha, you have a couple of options for use with Mojo. We just released MAX, which included a graph API where you can build out models. You can read more of the documentation about it here: https://docs.modular.com/engine/graph

We also have a bunch of community contributed AI libraries, including ML inference libraries, and you can find a list here: https://github.com/mojicians/awesome-mojo?tab=readme-ov-file#ai

Backpropagation isn't something that's widely supported yet, but for simple CNNs there are lots of examples for manually implementing it. It's been a while since you posted, but we'd love to hear about what you've been working on.
